src/components/Checkbox.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <label :for="id" class="checkbox-label" :class="labelClasses">
+    <input
+      :id="id"
+      type="checkbox"
+      class="checkbox"
+      :checked="checked"
+      :name="inputName"
+      :value="inputValue"
+      :disabled="disabled"
+      @change="onChange"
+    /><span>
+      <slot>
+        {{ label }}
+      </slot>
+    </span>
+  </label>
+</template>
+
+<script>
+export default {
+  name: 'Checkbox',
+  props: {
+    /**
+     * Checkbox `id` is used for the input id property, connecting the label to
+     * the checkbox. id is also used in the `change` event payload as the `name`
+     * and `value` parameters if they are not set.
+     */
+    id: {
+      type: String,
+      required: true,
+    },
+    /**
+     * The checkbox can be labeled either using the `label` prop, or the default
+     * slot (for labels that can contain more than simple strings).
+     *
+     * @default ''
+     */
+    label: {
+      type: String,
+      default: '',
+    },
```

Comment:
Having two ways to set the label seems confusing to me and also potentially prone to someone forgetting to add a label. What if we go with just the slot and somehow verify that it is passed? Is there a scenario where we wouldn't be able to pass the slot value and would have to use a prop instead?

Reply:
I've removed the label prop. I guess it was my first thought to add the label using prop, and then I just went with it without much thought. AND got confused with whether I need to pass a prop or not myself, so thank you for this comment!